sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -635,7 +637,8 @@ class SparkSession private(
    * @since 2.0.0
    */
   def sql(sqlText: String): DataFrame = {
-    Dataset.ofRows(self, sessionState.sqlParser.parsePlan(sqlText))
+    Dataset.ofRows(self, sessionState.sqlParser.parsePlan(sqlText),
+      substitutor.substitute(sqlText))
```

Comment:
Hi, @LantaoJin .
What you need is just grapping the initial SQL text here, you can use Spark extension. Please refer [Spark Atlas Connector](https://github.com/hortonworks-spark/spark-atlas-connector/blob/master/spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/sql/SparkExtension.scala) for a sample code.

Reply:
You may want to refactor this PR into `ParserExtension` and UI part. I think that will be less intrusive than the current implementation.

Reply:
BTW, in general, the initial SQL texts easily become meaningless when another operations are added. In your example, the following case shows a misleading and wrong SQL statement instead of *real executed SQL plan*.

```scala
val df = spark.sql("xxxxx")
df.filter(...).collect() // shows sql text "xxxxx"
```

As another example, please try the following. It will show you `select a,b from t1`.
```
scala> spark.sql("select a,b from t1").select("a").show
```

Reply:
>  the following case shows a misleading and wrong SQL statement instead of real executed SQL plan.

Yes. We know this, so current implementation which bind sql text to DF is not good.